### ISSUE TITLE
libobjc2 is out of date

### ISSUE THREAD

**Issue by davidchisnall** [ravynsoft/ravynos#432]
After a period where I thought we were done, we've had a flurry of development in libobjc2.  Features since your last import include:

 - Support for direct methods (needs a newer clang, I think you have a new-enough one).
 - RISC-V support
 - PowerPC support for block trampolines.
 - Fast-path alloc-family methods (mostly a code-size reduction, can be a speedup too)
 - The legacy GC stuff is gone.
 - Several bug fixes including a couple of data corruption and one deadlock

**Comment by mszoek** [comment]
Awesome! I'll pull the changes! We have clang 18 now... should be ok?

**Comment by davidchisnall** [comment]
Yup, 18 is fine. With older clang, noting breaks, but if you do `-fobjc-runtime=gnustep-2.1` you’ll get the 2.0 behaviour. The runtime is backwards compatible, so that just means the generated code will be (slightly) slower and larger, but it will still work.

**Comment by davidchisnall** [comment]
I’m not sure if you’re doing anything with Swift, but there’s also an open PR to (optionally) use the Swift blocks runtime instead of our own, which would probably help a bit if you are.

**Comment by mszoek** [comment]
Swift is definitely planned so that should be really helpful.
